### PR TITLE
feat: added provider backing service and a services refactor

### PR DESCRIPTION
**Pull request by tucker-weed**
Description

Provider backing service:
1. Doc awareness updates (central authority) backing service implemented

Refactor:
1. Creates two separate connection resolvers for the two web socket services
2. Factors out message resolvers for the two web socket services

### CODE REVIEW

**Comment by tucker-weed** [comment]
Many of these changes are not actual changes - I based this branch off of my other PR branch, so it's showing those changes as well.

File: src/editor.tsx [hunk 1]
```diff
@@ -8,22 +8,27 @@ import "codemirror/lib/codemirror.css";
 import { WebsocketProvider } from "y-websocket";
 
 const ROOM_NAME = "new-room";
-const SIGNALLING_SERVICE = process.env.STORYBOOK_SIGNAL_URL|| 'ws://localhost:6001';
+const SIGNALLING_SERVICE =
+  process.env.STORYBOOK_SIGNAL_URL || "ws://localhost:6001/signal";
+const PROVIDER_SERVICE =
+  process.env.STORYBOOK_PROVIDER_URL || "ws://localhost:6001/provider";
```
Comment (chewnoill) [inline src/editor.tsx:14]: needs a rebase. Default URLs should be `ws://localhost:6006/ws/...`

File: src/websocket-service.ts [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
+ *
  * Data persistance.
  *   The y-websocket interface implements a leveldb data layer interface.
  *   * What is leveldb?
  *   * Implement connector to some docker database.
  */
- 
 
-const host = process.env.HOST || '0.0.0.0'
+const host = process.env.HOST || "0.0.0.0";
```
Comment (chewnoill) [inline src/websocket-service.ts:29]: avoid touching lines you don't need to touch. If you need to make formatting or whitespace changes, keep those in their own commit.

Comment (tucker-weed) [reply src/websocket-service.ts:29]: Not really sure how that happened to be honest, I'll revert that change back.

File: src/utils/provider-connection.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * Setup a new client
+ * @param {any} conn
+ * @param {any} req
+ * @param {any} options
+ */
+const setupProviderConnection = (
+  conn,
+  req,
+  { docName = req.url.slice(1).split("/")[1], gc = true } = {}
```
Comment (chewnoill) [inline src/utils/provider-connection.ts:46]: I think there are a couple different refactors that are happening here, but its hard to see what they are because we are moving things around at the same time. It's better to keep these refactors on different commits so I can see whats changing.

On the other hand, its bad form to do refactors on things you just added. Let's squash these commits together and try to find a different way to break them apart.

Try to think about making your changes in steps:
1. [refactor] Move the signaling service to `/signal`
2. [feat] Add the provider servicer to `/provider`

Comment (tucker-weed) [reply src/utils/provider-connection.ts:46]: I'll get on that - would it be good to put the change of `{ docName = req.url.slice(1).split("?")[1], gc = true } = {}` to `{ docName = req.url.slice(1).split("/")[1], gc = true } = {}` on a separate [fix] commit? I don't think splitting on "?" is correct because there isn't that character in the url, so I changed it to "/" to get the room name. This kind of change warrants it's own commit I guess is what I'm asking?

File: src/utils/signaling-connection.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
+import * as map from "lib0/map";
```
Comment (chewnoill) [inline src/utils/signaling-connection.ts:1]: I was hoping that git would show this a file rename instead of an add and a delete. There is some minimum amount of changes to this file that is preventing that from happening.

**Review by chewnoill — APPROVED** [review]
one nit, but not blocking. Great work @tucker-weed
